server/src/test/java/org/opensearch/indices/replication/SegmentReplicationTargetTests.java

```diff
@@ -0,0 +1,256 @@
+import org.opensearch.indices.replication.common.ReplicationType;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class SegmentReplicationTargetTests extends IndexShardTestCase {
+
+    SegmentReplicationTarget segrepTarget;
+    IndexShard indexShard, spyIndexShard;
+    ReplicationCheckpoint repCheckpoint;
+    ByteBuffersDataOutput buffer;
+
+    private static final StoreFileMetadata SEGMENTS_FILE = new StoreFileMetadata(IndexFileNames.SEGMENTS, 1L, "0", Version.LATEST);
+    private static final StoreFileMetadata PENDING_DELETE_FILE = new StoreFileMetadata("pendingDelete.del", 1L, "1", Version.LATEST);
+
+    private static final Store.MetadataSnapshot SI_SNAPSHOT = new Store.MetadataSnapshot(
+        Map.of(SEGMENTS_FILE.name(), SEGMENTS_FILE),
+        null,
+        0
+    );
+
+    SegmentInfos testSegmentInfos = new SegmentInfos(Version.LATEST.major);
+
+    @Override
+    public void setUp() throws Exception {
+
+        super.setUp();
+        buffer = new ByteBuffersDataOutput();
+        try (ByteBuffersIndexOutput indexOutput = new ByteBuffersIndexOutput(buffer, "", null)) {
+            testSegmentInfos.write(indexOutput);
+        }
+
+        repCheckpoint = mock(ReplicationCheckpoint.class);
+        Settings indexSettings = Settings.builder()
+            .put(IndexMetadata.SETTING_VERSION_CREATED, org.opensearch.Version.CURRENT)
+            .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
+            .build();
+        indexShard = newStartedShard(false, indexSettings, new NRTReplicationEngineFactory());
+        spyIndexShard = spy(indexShard);
+        Mockito.doNothing().when(spyIndexShard).finalizeReplication(any(SegmentInfos.class), any(), anyLong());
+    }
+
+    public void testSuccessfulResponse_startReplication() {
+
+        SegmentReplicationSource segrepSource = new SegmentReplicationSource() {
+            @Override
+            public void getCheckpointMetadata(
+                long replicationId,
+                ReplicationCheckpoint checkpoint,
+                ActionListener<CheckpointInfoResponse> listener
+            ) {
+                listener.onResponse(new CheckpointInfoResponse(checkpoint, SI_SNAPSHOT, buffer.toArrayCopy(), Set.of(PENDING_DELETE_FILE)));
+            }
+
+            @Override
+            public void getSegmentFiles(
+                long replicationId,
+                ReplicationCheckpoint checkpoint,
+                List<StoreFileMetadata> filesToFetch,
+                Store store,
+                ActionListener<GetSegmentFilesResponse> listener
+            ) {
```

Comment:
we compute a diff and ask for files missing locally that are in SI_SNAPSHOT + PENDING_DELETE_FILES.

Lets add an assertion here on the files contained in `filesToFetch` ?